src/infra/components/jobs.ts

```diff
@@ -85,36 +141,42 @@ export class JobSystem extends Construct {
     // Create task definitions for each job type
     this.taskDefinitions = {};
 
-    for (const [jobType, config] of Object.entries(props.jobTypes)) {
-      const taskDef = new ecs.FargateTaskDefinition(
-        this,
-        `${jobType}-task-def`,
-        {
-          cpu: config.cpu,
-          memoryLimitMiB: config.memoryLimitMiB,
-        },
-      );
+    for (const workerConfig of props.workers) {
+      const jobId = workerConfig.jobTypes.join('-');
+      const taskDef = new ecs.FargateTaskDefinition(this, `${jobId}-task-def`, {
+        cpu: workerConfig.cpu,
+        memoryLimitMiB: workerConfig.memoryLimitMiB,
+      });
 
       // Grant task role access to S3 bucket
       this.storageBucket.grantReadWrite(taskDef.taskRole);
 
+      // Add efs config if necessary
+      if (workerConfig.efsMounts) {
+        // Add vols
+        for (const vol of workerConfig.efsMounts.volumes ?? []) {
+          taskDef.addVolume(vol);
+        }
+      }
+
       // Add container to task definition
-      taskDef.addContainer(`${jobType}-container`, {
+      const containerDfn = taskDef.addContainer(`${jobId}-container`, {
         // This specifies the image to be used - should be in the full format
         // i.e. "ghcr.io/open-aims/reefguideapi.jl/reefguide-src:latest"
-        image: ecs.ContainerImage.fromRegistry(config.workerImage),
+        image: ecs.ContainerImage.fromRegistry(workerConfig.workerImage),
         // Docker command
-        command: config.command,
+        command: workerConfig.command,
         logging: ecs.LogDrivers.awsLogs({
-          streamPrefix: `worker-${jobType.toLowerCase()}`,
+          streamPrefix: `worker-${jobId.toLowerCase()}`,
           logRetention: logs.RetentionDays.ONE_WEEK,
         }),
         environment: {
           API_ENDPOINT: props.apiEndpoint,
           AWS_REGION: Stack.of(this).region,
-          // TODO might be multiple job types here
-          JOB_TYPES: jobType,
+          JOB_TYPES: workerConfig.jobTypes.join(','),
           S3_BUCKET_NAME: this.storageBucket.bucketName,
+          // Custom additional environment variables
+          ...(workerConfig.env ?? {}),
```

Comment:
can't you just do `...workerConfig.env`? should be fine if it's null or undefined

Reply:
You can't spread an undefined value

Reply:
not important, but I'm down this JavaScript/TypeScript rabbit hole:
you can https://stackoverflow.com/a/62885880

but I just learned the type you're spreading must be `X | undefined`, not narrowed to `undefined`
https://github.com/microsoft/TypeScript/issues/55083#issuecomment-1642523713

This works:
```typescript
    const foo = undefined as object | undefined;
    const o = {
      a: 1,
      ...foo
    }
```